### PR TITLE
Fix: fmt::ostream cannot be moved while holding buffered data #2197

### PR DESCRIPTION
**Pull request by vtta**
#2197

### CODE REVIEW

File: include/fmt/os.h [hunk 1]
```diff
@@ -396,8 +396,10 @@ class ostream final : private detail::buffer<char> {
   ostream(ostream&& other)
       : detail::buffer<char>(other.data(), other.size(), other.capacity()),
         file_(std::move(other.file_)) {
+    other.clear();
     other.set(nullptr, 0);
```
Comment (vitaut) [inline include/fmt/os.h:400]: I wonder if this can be replaced with

```
other = detail::buffer<char>();
```

**Review by vitaut — COMMENTED** [review]
Thanks for the PR!

**Comment by alexezeder** [comment]
Hmm.... @vitaut, is a line in PR description:

> I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

still necessary?

File: include/fmt/os.h (hunk 1 above)
Comment (vtta) [reply include/fmt/os.h:400]: It seems reasonable, but I prefer the consistency between `set` and `clear`. 
Another issue is that creating another object in ctors seems confusing.

**Pull request merged**